cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/StructuredTable.java

```diff
@@ -84,6 +84,36 @@
    */
   CloseableIterator<StructuredRow> scan(Range keyRange, int limit) throws InvalidFieldException, IOException;
 
+  /**
+   * Atomically compare and swap the value of a column in a row if the expected value matches.
+   * To match a non-existent value, the value of the expected field should be null.
+   *
+   * @param keys the primary key of row
+   * @param oldValue the expected value in the table
+   * @param newValue the new value to write if the expected value matches
+   * @return true if the compare and swap was successful, false otherwise
+   * @throws InvalidFieldException if any of the keys/fields are not part of table schema,
+   *                               or their types do not match the schema
+   * @throws IOException if there is an error reading or writing to the table
+   * @throws IllegalArgumentException if the field name or the type of the oldValue and newValue do not match
```

Comment:
do we allow compareAndSwap of a primary key column?

Reply:
Good point, will update documentation